src/decorator_base.rb

```diff
@@ -0,0 +1,30 @@
+require './nameable'
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    if @nameable.correct_name.length > 10
+      @nameable.correct_name.slice(0, 10)
+    else
+      @nameable.correct_name
+    end
+  end
+end
```

Comment:
- [ ] [optional] @Diegogagan2587 Great work adding decoration to your school library, it would have been better if each class was in its own file, that way its easier to read and scale your project in future.

Reply:
@danmainah thanks for pointing that out, I'll make the correction!